bin/varnishd/cache/cache_fetch.c

```diff
@@ -770,15 +770,16 @@ vbf_objiterate(void *priv, unsigned flush, const void *ptr, ssize_t len)
 	uint8_t *pd;
 
 	CAST_OBJ_NOTNULL(bo, priv, BUSYOBJ_MAGIC);
-	(void)flush;
+
+	flush &= OBJ_ITER_FLUSH;
```

Comment:
Isn't it `OBJ_ITER_END` that is the flag to test for here? I guess the logic holds for 304 revalidations because there wouldn't be any VDPs in the mix. But if there was e.g. a VDP gunzip in the mix there, the `OBJ_ITER_FLUSH` would be set multiple times during iteration. This looks to me like it opens future gotchas.